phoenix-core/src/test/java/org/apache/phoenix/query/ConnectionQueryServicesImplTest.java

```diff
@@ -194,7 +194,28 @@ public void testGetNextRegionStartKey() {
         GlobalClientMetrics.GLOBAL_HBASE_COUNTER_METADATA_INCONSISTENCY.getMetric().reset();
         when(mockHRegionInfo.getStartKey()).thenReturn(notCorruptedStartKey);
         when(mockHRegionInfo.getEndKey()).thenReturn(notCorruptedNewKey);
-        testGetNextRegionStartKey(mockCqsi, mockRegionLocation, notCorruptedEndKey, false);
+        testGetNextRegionStartKey(mockCqsi, mockRegionLocation, notCorruptedEndKey, true);
```

Comment:
Why is isCorrupted true here if both start and endkey are not corrupt and the current endkey is greater than the previous end key?

Reply:
for the new region, the startKey in this case is 0x2999 and endKey is 0x3001. The previous regions end key is 0x3000 leading to an overlap. Hence, isCorrupted is true here. However, your confusion makes sense because the variable names were incorrect. I have updated the variable names to start with "corrupted" in all cases where isCorrupted is true and "notCorrupted" otherwise. I have pushed with new changes. Please review!